### ISSUE TITLE
Can't disable reachability analysis

### ISSUE THREAD

**Issue by saagarjha** [sublimelsp/LSP-pyright#388]
I'm trying to set Pyright's `enableReachabilityAnalysis` setting, but it doesn't seem to work. I have this in my LSP-pyright.sublime-settings file:

```json
// Settings in here override those in "LSP-pyright/LSP-pyright.sublime-settings"
{
	"settings": {
		"python.analysis.diagnosticSeverityOverrides": {
			"enableReachabilityAnalysis": false,
		},
	}
}
```

I would assume this is the right place for it? But even after setting this I'm still getting diagnostics for it:

<img width="415" alt="Image" src="https://github.com/user-attachments/assets/ac5d7636-8142-4e95-8fc5-f07891e878c7" />

Did I put the setting in the right place? Or, perhaps there is a way to debug if these settings are being passed through? Thanks!

**Comment by jfcherng** [comment]
Either an upstream issue or expected. See upstream's explanation on https://github.com/microsoft/pyright/blob/main/docs/configuration.md

> enableReachabilityAnalysis [boolean]: If enabled, code that is determined to be unreachable by type analysis is reported using a tagged hint. This setting does not affect code that is determined to be unreachable regardless of type analysis; such code is always reported as unreachable. This setting also has no effect when using the command-line version of pyright because it never emits tagged hints for unreachable code.

**Comment by jfcherng** [comment]
It seems that `enableReachabilityAnalysis` is for `pyrightconfig.json` only 🤔 

I don't see it in https://github.com/microsoft/pyright/blob/54f7da25f9c2b6253803602048b04fe0ccb13430/packages/vscode-pyright/package.json

**Comment by saagarjha** [comment]
Hmm, so there's no way to actually set this in the Sublime settings file? Should I file a bug upstream asking them to add it?

**Comment by jfcherng** [comment]
Those are upstream's design decision. Nothing we can do here.

**Comment by saagarjha** [comment]
Ok, I filed https://github.com/microsoft/pyright/issues/10239. Thanks for your help!

**Comment by saagarjha** [comment]
Actually, while you're here, do you know if any changes went in for displaying hints? For Pyright this analysis is one of the few things that get reported as hints, so by turning those off I can effectively get the same effect. At one point I was able to turn them off, I think by setting `"pyright.disableTaggedHints": true`. But as of this week they're back and I'm not entirely sure this was supposed to do anything anyway.

**Comment by rchl** [comment]
There is an LSP setting for controlling which level of diagnostics you want to see in the view - `show_diagnostics_severity_level`. By default hint-level diagnostics are enabled.
